hazelcast/include/hazelcast/client/connection/ClientConnectionManagerImpl.h

```diff
@@ -204,22 +216,27 @@ namespace hazelcast {
                 HeartbeatManager heartbeat_;
                 std::vector<std::thread> io_threads_;
                 std::unique_ptr<boost::asio::io_context::work> io_guard_;
-                std::atomic<int32_t> partition_count_;
                 const bool async_start_;
                 const config::client_connection_strategy_config::reconnect_mode reconnect_mode_;
                 const bool smart_routing_enabled_;
                 boost::optional<boost::asio::steady_timer> connect_to_members_timer_;
-                std::atomic_bool connect_to_cluster_task_submitted_;
                 boost::uuids::uuid client_uuid_;
                 boost::chrono::milliseconds authentication_timeout_;
+                std::vector<std::string> labels_;
+                load_balancer &load_balancer_;
+                wait_strategy wait_strategy_;
+
+                // following fields are updated inside synchronized(clientStateMutex)
+                std::mutex client_state_mutex_;
+                util::SynchronizedMap<boost::uuids::uuid, Connection, boost::hash<boost::uuids::uuid>> active_connections_;
+                util::SynchronizedMap<int32_t, Connection> active_connection_ids_;
```

Comment:
Now that the accesses to `active_connections_` and `active_connection_ids_` are protected with the mutex `client_state_mutex_`, can they be regular `unordered_map`s instead of `util::SynchronizedMap`?

Reply:
There are still methods such as `get_connection(boost::uuids::uuid uuid)` and `get_connection(const address &address)` which do not use the `client_state_mutex_`, hence we can not completely use regular maps.